src/main/java/org/spongepowered/api/item/recipe/RecipeRegistry.java

```diff
@@ -24,32 +24,32 @@
  */
 package org.spongepowered.api.item.recipe;
 
-import java.util.Set;
+import java.util.Collection;
 
 /**
  * A RecipeRegistry holds all registered recipes for a given game.
  */
-public interface RecipeRegistry {
+public interface RecipeRegistry<T extends Recipe> {
 
     /**
-     * Registers the given Recipe to make it available to craft.
+     * Registers the given CraftingRecipe to make it available to craft.
      *
-     * @param recipe The Recipe to register
+     * @param recipe The CraftingRecipe to register
      */
-    void register(Recipe recipe);
+    void register(T recipe);
 
     /**
-     * Removes the given Recipe from registration in this registry.
+     * Removes the given CraftingRecipe from registration in this registry.
      *
-     * @param recipe The Recipe to unregister
+     * @param recipe The CraftingRecipe to unregister
```

Comment:
The javadocs mention crafting recipes but the generic type allows plain recipes as well.